File: 23_12_week3/wangon/1793.py
# 타일링
dp = ["1"] * 251

def add_num(a, b):
    ret = ""
    carry = 0
    i = len(a) - 1
    j = len(b) - 1
    while i >= 0 or j >= 0 or carry:
        _sum = carry
        if i >= 0:
            _sum += int(a[i])
            i -= 1
        if j >= 0:
            _sum += int(b[j])
            j -= 1
        carry = _sum // 10
        ret = str(_sum % 10) + ret
    return ret

for i in range(2, 251):
    dp[i] = add_num(dp[i-1], add_num(dp[i-2], dp[i-2]))

while True:
    try:
        n = int(input())
        print(dp[n])
    except EOFError:
        break

File: src/AS_23_12_week3/wangon/1793.cpp
// 타일링
#include <algorithm>
#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>
using namespace std;

string dp[251];

string addNum(string a, string b)
{
	string ret;
	int carry = 0;
	int i = a.length() - 1;
	int j = b.length() - 1;
	while (i >= 0 || j >= 0 || carry)
	{
		int sum = carry;
		if (i >= 0)
		{
			sum += a[i] - '0';
			i--;
		}
		if (j >= 0)
		{
			sum += b[j] - '0';
			j--;
		}
		carry = sum / 10;
		ret = to_string(sum % 10) + ret;
	}
	return ret;
}

int main()
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	dp[0] = "1";
	dp[1] = "1";
	for (int i = 2; i <= 250; i++)
		dp[i] = addNum(dp[i - 1], (addNum(dp[i - 2], dp[i - 2])));
	int n;
	while (cin >> n)
		cout << dp[n] << endl;
}
